### PR TITLE
Tuples: Adding value tuples 1 and 8, and some tuple parsing tests

### PR DESCRIPTION
**Pull request by jcouv**
From discussion with @VSadov, we think we'll stick with the current `System.Tuple` design with regards to tuples over 7. This implies that we do need a 1-tuple type, even though the language won't allow to produce one.

I needed to push the `WellKnownTypes` enum beyond 255 to add those two types. So I updated the byte stream representation of well-known type members to allow extended types, as discussed with @AlekseyTs.  
When the byte for a type is 255, it is followed by an extension byte, which can be added together.

Note that the existing `AllWellKnownTypeMembers` test calls `GetWellKnownTypeMember` on `System_String__Format_IFormatProvider`, thereby providing coverage for parts of the new logic.

CC @dotnet/roslyn-compiler for code review.

### CODE REVIEW

File: src/Compilers/Core/Portable/SignatureComparer.cs [hunk 1]
```diff
@@ -211,6 +212,25 @@ private bool MatchType(TypeSymbol type, ImmutableArray<byte> signature, ref int
         }
 
         /// <summary>
+        /// Read a type Id from the signature.
+        /// This may consume one or two bytes, and therefore increment the position correspondingly.
+        /// </summary>
+        private short ReadTypeId(ImmutableArray<byte> signature, ref int position)
```
Comment (cston) [inline src/Compilers/Core/Portable/SignatureComparer.cs]: `static`?

File: src/Compilers/Core/Portable/WellKnownMembers.cs [hunk 2]
```diff
@@ -2681,13 +2687,61 @@ static WellKnownMembers()
                 0,                                                                                                          // Arity
                     (byte)SignatureTypeCode.GenericTypeParameter, 6,                                                        // Field Signature
 
+                // System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item1
+                (byte)MemberFlags.Field,                                                                                    // Flags
+                (byte)WellKnownType.ExtendedType, WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest.ExtensionByte(), // DeclaringTypeId
```
Comment (cston) [inline src/Compilers/Core/Portable/WellKnownMembers.cs]: Does ```value.ExtensionByte()``` need to be a compile-time constant so the ```byte[]``` is stored as raw bytes in the assembly?

Comment (jcouv) [inline src/Compilers/Core/Portable/WellKnownMembers.cs]: You're correct. There was an impact on the generated IL for the constructor. 
But it seems that most bytes are still stored statically. Only a few are produced at runtime by the compiler with `ExtensionByte`.
I will try the same thing replacing ExtensionByte with (byte)(x-255) to compare the result.

Before:
```
.method private hidebysig specialname rtspecialname static 
	void .cctor () cil managed 
{
	// Method begins at RVA 0x41a68
	// Code size 3187 (0xc73)
	.maxstack 4
	.locals init (
		[0] uint8[],
		[1] string[]
	)

	IL_0000: ldc.i4 2700
	IL_0005: newarr [System.Runtime]System.Byte
	IL_000a: dup
	IL_000b: ldtoken field valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=2700' '<PrivateImplementationDetails>'::EC51BC739CB9D330F73835E6C77AB1F5BF232A31
	IL_0010: call void [System.Runtime]System.Runtime.CompilerServices.RuntimeHelpers::InitializeArray(class [System.Runtime]System.Array, valuetype [System.Runtime]System.RuntimeFieldHandle)
	IL_0015: stloc.0
	IL_0016: ldc.i4 294
	IL_001b: newarr [System.Runtime]System.String
	IL_0020: dup
	IL_0021: ldc.i4.0
	IL_0022: ldstr "Round"
	IL_0027: stelem.ref
	IL_0028: dup
	IL_0029: ldc.i4.1
	IL_002a: ldstr "Pow"
	IL_002f: stelem.ref
	IL_0030: dup
	IL_0031: ldc.i4.2
```



After:
```
.method private hidebysig specialname rtspecialname static 
	void .cctor () cil managed 
{
	// Method begins at RVA 0x57118
	// Code size 3773 (0xebd)
	.maxstack 4
	.locals init (
		[0] uint8[],
		[1] string[]
	)

	IL_0000: nop
	IL_0001: ldc.i4 2889
	IL_0006: newarr [System.Runtime]System.Byte
	IL_000b: dup
	IL_000c: ldtoken field valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=2889' '<PrivateImplementationDetails>'::'82FE9269DACA81A6359D20D1F2A5AF0913A6051B'
	IL_0011: call void [System.Runtime]System.Runtime.CompilerServices.RuntimeHelpers::InitializeArray(class [System.Runtime]System.Array, valuetype [System.Runtime]System.RuntimeFieldHandle)
	IL_0016: dup
	IL_0017: ldc.i4 2829
	IL_001c: ldc.i4 256
	IL_0021: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_0026: stelem.i1
	IL_0027: dup
	IL_0028: ldc.i4 2835
	IL_002d: ldc.i4 256
	IL_0032: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_0037: stelem.i1
	IL_0038: dup
	IL_0039: ldc.i4 2841
	IL_003e: ldc.i4 256
	IL_0043: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_0048: stelem.i1
	IL_0049: dup
	IL_004a: ldc.i4 2847
	IL_004f: ldc.i4 256
	IL_0054: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_0059: stelem.i1
	IL_005a: dup
	IL_005b: ldc.i4 2853
	IL_0060: ldc.i4 256
	IL_0065: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_006a: stelem.i1
	IL_006b: dup
	IL_006c: ldc.i4 2859
	IL_0071: ldc.i4 256
	IL_0076: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_007b: stelem.i1
	IL_007c: dup
	IL_007d: ldc.i4 2865
	IL_0082: ldc.i4 256
	IL_0087: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_008c: stelem.i1
	IL_008d: dup
	IL_008e: ldc.i4 2871
	IL_0093: ldc.i4 256
	IL_0098: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_009d: stelem.i1
	IL_009e: dup
	IL_009f: ldc.i4 2883
	IL_00a4: ldc.i4 257
	IL_00a9: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_00ae: stelem.i1
	IL_00af: stloc.0
	IL_00b0: ldc.i4 330
	IL_00b5: newarr [System.Runtime]System.String
	IL_00ba: dup
	IL_00bb: ldc.i4.0
	IL_00bc: ldstr "Round"
	IL_00c1: stelem.ref
	IL_00c2: dup
	IL_00c3: ldc.i4.1
	IL_00c4: ldstr "Pow"
	IL_00c9: stelem.ref
	IL_00ca: dup
	IL_00cb: ldc.i4.2
```

Comment (jcouv) [inline src/Compilers/Core/Portable/WellKnownMembers.cs]: @cston, Inlining `(byte)(x-255)` indeed brings the generated IL back to original state. I'll make that change.

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs [hunk 3]
```diff
@@ -77,7 +77,7 @@ internal Symbol GetWellKnownTypeMember(WellKnownMember member)
 
         internal NamedTypeSymbol GetWellKnownType(WellKnownType type)
         {
-            Debug.Assert(type >= WellKnownType.First && type <= WellKnownType.Last);
+            Debug.Assert(type >= WellKnownType.First && type <= WellKnownType.Last && type != WellKnownType.ExtendedType);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs]: There is a similar assert in VB.

File: src/Compilers/CSharp/Test/Symbol/Symbols/MissingSpecialMember.cs [hunk 4]
```diff
@@ -643,5 +662,18 @@ public void AllWellKnownTypeMembers()
                 Assert.NotNull(symbol);
             }
         }
+
+        [Fact]
+        public void FirstTypeAfterExtensionMarker()
+        {
+            var e = WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest.ExtensionByte();
+            Assert.Equal(1, e);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/MissingSpecialMember.cs]: I am not sure if there is a value in this test, values in an enum are auto incremented. We just need to test that WellKnownType.ExtendedType == 255, which can be done with an assert in static constructor of WellKnownTypes.

File: src/Compilers/CSharp/Test/Symbol/Symbols/MissingSpecialMember.cs [hunk 5]
```diff
@@ -643,5 +662,18 @@ public void AllWellKnownTypeMembers()
...
+        }
+
+        [Fact]
+        public void ExtensionMarkerProperlyAligned()
+        {
+            Assert.Equal(WellKnownType.ExtendedType, WellKnownTypes.GetTypeFromMetadataName("ExtendedType"));
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/MissingSpecialMember.cs]: Since it is not a real type, it shouldn't have a name, I think. Also, I think an assert in static constructor of WellKnownTypes should be sufficient. In any case this test doesn't belong in CSharp tests, it is not a language specific.

File: src/Compilers/Core/Portable/MemberDescriptor.cs [hunk 6]
```diff
@@ -1,10 +1,10 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: What changed on this line?

File: src/Compilers/Core/Portable/MemberDescriptor.cs [hunk 7]
```diff
@@ -114,6 +114,27 @@ internal static ImmutableArray<MemberDescriptor> InitializeFromStream(Stream str
...
+        /// </summary>
+        private static short ReadTypeId(Stream stream)
+        {
+            var firstByte = (byte)stream.ReadByte();
+
+            if (firstByte == (byte)WellKnownType.ExtendedType)
+            {
+                short result = (short)stream.ReadByte();
+                result += (short)WellKnownType.ExtendedType;
+                return result;
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: ```return (short)(WellKnownType.ExtendedType + stream.ReadByte()); ``` and remove the ```result``` local?

File: src/Compilers/Core/Portable/MemberDescriptor.cs [hunk 8]
```diff
@@ -142,6 +163,9 @@ private static void ParseType(ImmutableArray<byte>.Builder builder, Stream strea
                         throw ExceptionUtilities.UnexpectedValue(typeCode);
 
                     case SignatureTypeCode.TypeHandle:
+                        ParseTypeId(builder, stream);
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: Consider renaming to "ParseTypeHandle" to be consistent with ParseGenericTypeInstance.

File: src/Compilers/Core/Portable/MemberDescriptor.cs [hunk 9]
```diff
@@ -166,6 +190,27 @@ private static void ParseType(ImmutableArray<byte>.Builder builder, Stream strea
...
+        /// This may copy one or two bytes depending on the first one.
+        /// </summary>
+        private static void ParseTypeId(ImmutableArray<byte>.Builder builder, Stream stream)
+        {
+            var firstByte = (byte)stream.ReadByte();
+
+            if (firstByte == (byte)WellKnownType.ExtendedType)
+            {
+                var secondByte = (byte)stream.ReadByte();
+                builder.Add(firstByte);
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: Move this out of the if.

File: src/Compilers/Core/Portable/MemberDescriptor.cs [hunk 10]
```diff
@@ -166,6 +190,27 @@ private static void ParseType(ImmutableArray<byte>.Builder builder, Stream strea
...
+        /// Read a type Id from the stream and copy it into the builder.
+        /// This may copy one or two bytes depending on the first one.
+        /// </summary>
+        private static void ParseTypeId(ImmutableArray<byte>.Builder builder, Stream stream)
+        {
+            var firstByte = (byte)stream.ReadByte();
+
+            if (firstByte == (byte)WellKnownType.ExtendedType)
+            {
+                var secondByte = (byte)stream.ReadByte();
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: The local is unnecessary.

File: src/Compilers/Core/Portable/MemberDescriptor.cs [hunk 11]
```diff
@@ -166,6 +190,27 @@ private static void ParseType(ImmutableArray<byte>.Builder builder, Stream strea
...
+                builder.Add(firstByte);
+                builder.Add(secondByte);
+            }
+            else
+            {
+                builder.Add(firstByte);
+                return;
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: Unnecessary and is inconsistent with the 'if' block.

File: src/Compilers/Core/Portable/SignatureComparer.cs [hunk 12]
```diff
@@ -1,5 +1,6 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/SignatureComparer.cs]: What is changed on this line?

File: src/Compilers/Core/Portable/SignatureComparer.cs [hunk 13]
```diff
@@ -211,6 +212,25 @@ private bool MatchType(TypeSymbol type, ImmutableArray<byte> signature, ref int
...
+        /// This may consume one or two bytes, and therefore increment the position correspondingly.
+        /// </summary>
+        private short ReadTypeId(ImmutableArray<byte> signature, ref int position)
+        {
+            var firstByte = signature[position++];
+            if (firstByte == (byte)WellKnownType.ExtendedType)
+            {
+                short result = (short)signature[position++];
+                result += (short)WellKnownType.ExtendedType;
+                return result;
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/SignatureComparer.cs]: ```return (short)(WellKnownType.ExtendedType + signature[position++]);```?

File: src/Compilers/Core/Portable/WellKnownMembers.cs (hunk 2 above)
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/WellKnownMembers.cs]: Consider declaring a constant for this value, it is used in many places.

File: src/Compilers/Core/Portable/WellKnownTypes.cs [hunk 14]
```diff
@@ -252,12 +253,26 @@ internal enum WellKnownType
         System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6,
         System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7,
 
+        ExtendedType = 255, // Not a real type, just a marker for types above 255 and strictly below 512
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/WellKnownTypes.cs]: Don't assign explicit value to this member. Assert that it is 255 in static constructor of WellKnownTypes.

File: src/Compilers/Core/Portable/WellKnownTypes.cs [hunk 15]
```diff
@@ -252,12 +253,26 @@ internal enum WellKnownType
...
+
         System_IFormatProvider,
 
         Available,
         Last = Available - 1,
     }
 
+    internal static class WellKnownTypeExtensions
+    {
+        public static byte ExtensionByte(this WellKnownType type)
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/WellKnownTypes.cs]: It looks like, given the other feedback, this method is not going to be used. Consider removing.

File: src/Compilers/Core/Portable/WellKnownTypes.cs (hunk 14 above)
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/WellKnownTypes.cs]: We can also assert that Last < 512.

File: src/Compilers/Core/Portable/WellKnownTypes.cs [hunk 16]
```diff
@@ -501,6 +517,10 @@ internal static class WellKnownTypes
             "System.Runtime.CompilerServices.ValueTuple`6",
             "System.Runtime.CompilerServices.ValueTuple`7",
 
+            "ExtendedType",
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/WellKnownTypes.cs]: This is not a real type, try if we can use null. Use an empty string otherwise.

File: src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/WellKnownTypeValidationTests.vb [hunk 17]
```diff
@@ -499,15 +499,20 @@ End Namespace
...
                         Continue For
                     Case WellKnownType.System_FormattableString,
-                         WellKnownType.System_Runtime_CompilerServices_FormattableStringFactory,
-                         WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2,
-                         WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3,
-                         WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4,
-                         WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5,
-                         WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6,
-                         WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7
+                        WellKnownType.System_Runtime_CompilerServices_FormattableStringFactory,
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/WellKnownTypeValidationTests.vb]: Please restore original indentation.

**Comment by AlekseyTs** [comment]
@jcouv I am done with the review pass.

File: src/Compilers/Core/Portable/MemberDescriptor.cs (hunk 7 above)
Comment (VSadov) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: I'd hope local optimizer will elide this local anyways (in /o+) . 
My guess it is here for debugging.

File: src/Compilers/Core/Portable/MemberDescriptor.cs (hunk 6 above)
Comment (jcouv) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: Removed unicode signature &lt;U+FEFF> (it appears in some files...).

File: src/Compilers/Core/Portable/MemberDescriptor.cs (hunk 7 above)
Comment (jcouv) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: @AlekseyTs, I cannot add two bytes (it would overflow) or add two shorts with the `+` operator. That's why I used this longer way.

File: src/Compilers/Core/Portable/MemberDescriptor.cs (hunk 10 above)
Comment (jcouv) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: If it's ok, I'll keep this one for readability.

File: src/Compilers/Core/Portable/WellKnownTypes.cs (hunk 15 above)
Comment (jcouv) [inline src/Compilers/Core/Portable/WellKnownTypes.cs]: Correct. I'll remove.

**Comment by jcouv** [comment]
@cston, @AlekseyTs, the latest commit should address your feedback. 
Aleksey mentioned the concept of sentinel, which I thought is a good description, so I renamed `ExtendedType` to `ExtSentinel` in passing (since it was never a type...).

File: src/Compilers/CSharp/Test/Symbol/Symbols/MissingSpecialMember.cs [hunk 18]
```diff
@@ -634,6 +643,16 @@ public void AllWellKnownTypeMembers()
...
+                    case WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item1:
+                    case WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item2:
+                    case WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item3:
+                    case WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item4:
+                    case WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item5:
+                    case WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item6:
+                    case WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item7:
+                    case WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Rest:
+                        // PROTOTYPE tuples
                         // Not available yet, but will be in upcoming release.
```
Comment (cston) [inline src/Compilers/CSharp/Test/Symbol/Symbols/MissingSpecialMember.cs]: Consider changing comment to match line 569.

File: src/Compilers/CSharp/Test/Syntax/Parsing/ValueTupleTests.cs [hunk 19]
```diff
@@ -0,0 +1,494 @@
+using Xunit;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Parsing
+{
+    class ValueTupleTests : ParsingTests
```
Comment (cston) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ValueTupleTests.cs]: Do the tests need to be `public`?

File: src/Compilers/Core/Portable/MemberDescriptor.cs [hunk 20]
```diff
@@ -72,7 +72,7 @@ public string DeclaringTypeMetadataName
 
         public MemberDescriptor(
             MemberFlags Flags,
-            byte DeclaringTypeId,
+            short DeclaringTypeId,
             string Name,
             ImmutableArray<byte> Signature,
             ushort Arity = 0)
```
Comment (cston) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: Not part of this change but consider changing parameter names to have lowercase first character.

File: src/Compilers/Core/Portable/MemberDescriptor.cs [hunk 21]
```diff
@@ -114,6 +114,25 @@ internal static ImmutableArray<MemberDescriptor> InitializeFromStream(Stream str
             return builder.ToImmutable();
         }
 
+        /// <summary>
+        /// The type Id may be:
+        ///     (1) encoded in a single byte (for types below 254)
```
Comment (cston) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: `(for types below 255)`?

**Comment by cston** [comment]
LGTM

File: src/Compilers/CSharp/Test/Syntax/Parsing/ValueTupleTests.cs (hunk 19 above)
Comment (jcouv) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ValueTupleTests.cs]: Good question. It didn't have trouble running from TestDriven.Net, but I didn't verify they were run from xUnit.

File: src/Compilers/Core/Portable/MemberDescriptor.cs (hunk 20 above)
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: @jcouv If you decide to do the rename, please do it in a separate commit.

Comment (jcouv) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: @AlekseyTs Yes, I'll hold off on renaming to lowercase. Good to go other than that?

File: src/Compilers/CSharp/Test/Semantic/Semantics/ValueTupleTests.cs [hunk 22]
```diff
@@ -14,6 +14,10 @@ public void TestWellKnownMembersForValueTuple()
             var source = @"
 namespace System.Runtime.CompilerServices
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ValueTupleTests.cs]: I suspect these will end up directly in System.

File: src/Compilers/Core/Portable/MemberDescriptor.cs (hunk 6 above)
Comment (gafter) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: That is the unicode byte-order mark. It helps some editors figure out how to interpret the contents of the file. It doesn't need to be deleted.

**Comment by gafter** [comment]
LGTM

**Comment by jcouv** [comment]
@AlekseyTs Kindly review the PR feedback commit above.  
@cston Latest commit includes your suggestions.

File: src/Compilers/CSharp/Test/Syntax/Parsing/ValueTupleTests.cs [hunk 23]
```diff
@@ -0,0 +1,494 @@
+using Xunit;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ValueTupleTests.cs]: Copyright is missing.

File: src/Compilers/Core/Portable/MemberDescriptor.cs (hunk 6 above)
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/MemberDescriptor.cs]: I guess our default encoding for source is UTF-8.

**Comment by AlekseyTs** [comment]
LGTM, modulo the missing copyright in the new test file.

File: src/Compilers/CSharp/Test/Semantic/Semantics/ValueTupleTests.cs [hunk 24]
```diff
@@ -59,8 +63,22 @@ struct ValueTuple<T1, T2, T3, T4, T5, T6, T7>
...
+    struct ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest>
+    {
+        public T1 Item1;
+        public T2 Item2;
+        public T3 Item3;
+        public T4 Item4;
+        public T5 Item5;
+        public T6 Item6;
+        public T7 Item7;
+        publid TRest Rest;
```
Comment (apskim) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ValueTupleTests.cs]: public?

**Comment by jcouv** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/5258/
// Retest reason: known issue being investigated

**Comment by gafter** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/5258/
// Retest reason: timed out
